src/main/scala/async/Listener.scala

```diff
@@ -1,73 +1,178 @@
 package gears.async
 
-import gears.async.Listener.Locked
-import gears.async.Listener.Failed
-import gears.async.Listener.SemiLock
 import scala.annotation.tailrec
 import gears.async.Async.Source
+import java.util.concurrent.locks.ReentrantLock
+
+/** A listener, representing an one-time value receiver of an [[Async.Source]].
+  *
+  * Most of the time listeners should involve only calling a receiver function,
+  * and can be created by [[Listener.apply]] or [[Listener.acceptingListener]].
+  *
+  * However, should the listener want to attempt synchronization, it has to
+  * expose some locking-related interfaces. See `lock` and `release`.
```

Comment:
Release is gone (moved to lock)
```suggestion
  * expose some locking-related interfaces. See `lock`.
```